test/object/subscript_operator.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+const buildType = process.config.target_defaults.default_configuration;
+const assert = require('assert');
+
+test(require(`../build/${buildType}/binding.node`));
+test(require(`../build/${buildType}/binding_noexcept.node`));
+
+function test(binding) {
+  function testProperty(obj, key, value, nativeGetProperty, nativeSetProperty) {
+    nativeSetProperty(obj, key, value);
+    assert.strictEqual(nativeGetProperty(obj, key), value);
+  }
+
+  testProperty({}, 'key', 'value', binding.object.subscriptGetWithCStyleString, binding.object.subscriptSetWithCStyleString);
+  testProperty({ key: 'override me' }, 'key', 'value', binding.object.subscriptGetWithCppStyleString, binding.object.subscriptSetWithCppStyleString);
+  testProperty({}, 0, 'value', binding.object.subscriptGetAtIndex, binding.object.subscriptSetAtIndex);
```

Comment:
Could you add:
```js
testProperty({key: 'override me'}, 0, 'value', binding.object.subscriptGetAtIndex, binding.object.subscriptSetAtIndex);
```

Reply:
Of course, added in https://github.com/nodejs/node-addon-api/pull/923/commits/8e2ee12daedff316f3c0ff1316bae820b7c375cb